src/MongoDB.Driver.Core/Core/Clusters/MultiServerCluster.cs

```diff
@@ -43,6 +41,7 @@ internal sealed class MultiServerCluster : Cluster
         private readonly List<IClusterableServer> _servers;
         private readonly object _serversLock = new object();
         private readonly InterlockedInt32 _state;
+        private readonly object _updateClusterDescriptionLock = new object();
```

Comment:
We didn't need this lock before because UpdateClusterDescription was only called from the background task that processed the server description changed event queue, meaning there didn't used to be any chance of conflict. But now we also call UpdateClusterDescription in response to calls from the DnsMonitor thread.

Reply:
Ack